bigtop-deploy/puppet/modules/phoenix/manifests/init.pp

```diff
@@ -0,0 +1,36 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+class phoenix {
+
+  class deploy ($roles) {
+    if ('phoenix-server' in $roles) {
+      include phoenix::server
+    }
+  }
+
+  class server {
+    package { 'phoenix':
+      ensure => latest,
+      require => Package["hbase"],
+    }
+    file { 'phoenix-server-lib': 
+      path => '/usr/lib/hbase/lib/phoenix-server.jar',
+      target => '/usr/lib/phoenix/phoenix-server.jar',
+      ensure => link,
+      require => Package['phoenix'],
+    }
```

Comment:
How about adding `notify` to restart hbase-master and hbase-regionserver after the link is created?

Reply:
hi, @iwasakims, thanks for you advice. only there are some file changes, the relationship metaparameter **notify** will notify target resource to refresh. However, the phoenix-server-lib file resource operate a jar file, so we seldom edit it as we edit a text file or config file, the file maybe never changes, and then never refresh target resource(restart service). we can use **exec** resource file to implement the **restart** tasks.